File: PersonnelDepartment/Domain/Departments/DepartmentStructure.cs
using PersonnelDepartment.Domain.Posts;

namespace PersonnelDepartment.Domain.Departments;

public class DepartmentStructure
{
    public Department Department { get; }
    public Post[] Posts { get; }

    public DepartmentStructure(Department department, Post[] posts)
    {
        Department = department;
        Posts = posts;
    }
}

File: PersonnelDepartment/ClientApp/src/domain/departments/departmentStructure.ts
import { mapToPosts, Post } from "../posts/post";
import { Department, mapToDepartment } from "./department";

export class DepartmentStructure {
    constructor(
        public department: Department,
        public posts: Post[]
    ) { }
}

export function mapToDepartmentStructure(data: any): DepartmentStructure {
    const department = mapToDepartment(data.department);
    const posts = mapToPosts(data.posts);

    return new DepartmentStructure(department, posts);
}

export function mapToDepartmentStructures(data: any[]): DepartmentStructure[] {
    return data.map(mapToDepartmentStructure);
}